Include data for BCG stellar-halo mass relation

FHusko opened the pull request:
This data includes observational measurements of BCG stellar masses in fixed (2D) apertures of 10, 30, 50 and 100 kpc, so they can be used to compare with SOAP outputs. Various conversions are described in the files.

EvgeniiChaikin commented:
Hi @FHusko, thanks for this nice update!

For each paper in your dataset, can you please move multiple scripts that use different apertures to convert data into single files, such that one paper/author == one script/file? Then there will be ~3 times less code, and there will be less repeated code.

EvgeniiChaikin commented:
Then in each script, you can write something like

```
for aperture in [10, 50, 100]:
   do ...
   ... convert ...
   output_filename = "AuthorYear_{aperture}kpc.hdf5"
   output_path = f"{output_directory}/{output_filename}"
   processed.write(filename=output_path)
```

FHusko commented:
Thanks @EvgeniiChaikin for the comments! I have now done this, it should be much more condensed.

EvgeniiChaikin commented on data/GalaxyStellarMassHaloMassBCG/conversion/convertDeMaio2020.py:
```diff
@@ -0,0 +1,70 @@
+redshift_upper = 0.5
+h = h_sim
+
+# Write a separate file for each aperture
+apertures = [10, 50, 100]
+for x in range(3):
+    output_filename = f"DeMaio2020_{apertures[x]}.hdf5"
+    output_directory = "../"
+
+    if not os.path.exists(output_directory):
+        os.mkdir(output_directory)
```
```suggestion
output_directory = "../"
if not os.path.exists(output_directory):
    os.mkdir(output_directory)

apertures = [10, 50, 100]  # kpc

# Write a separate file for each aperture
for x, aperture in enumerate(apertures):
    output_filename = f"DeMaio2020_{aperture}kpc.hdf5"
```

EvgeniiChaikin commented on data/GalaxyStellarMassHaloMassBCG/conversion/convertGoldenMarx2023.py:
```diff
@@ -0,0 +1,85 @@
+redshift_upper = 0.8
+h = h_sim
+
+# Write a separate file for each aperture
+apertures = [10, 30, 50, 100]
+for x in range(4):
+
+    output_filename = f"GoldenMarx2023_{apertures[x]}.hdf5"
+    output_directory = "../"
+
+    if not os.path.exists(output_directory):
+        os.mkdir(output_directory)
```
```suggestion
output_directory = "../"
if not os.path.exists(output_directory):
    os.mkdir(output_directory)

apertures = [10, 30, 50, 100]  # kpc
    
# Write a separate file for each aperture
for x, aperture in enumerate(apertures):
    output_filename = f"GoldenMarx2023_{aperture}kpc.hdf5"
```

EvgeniiChaikin commented on data/GalaxyStellarMassHaloMassBCG/conversion/convertHuang2022.py:
```diff
@@ -0,0 +1,86 @@
+redshift_upper = 0.5
+h = h_sim
+
+# Write a separate file for each aperture
+apertures = [10, 30, 50, 100]
+for x in range(4):
+
+    output_filename = f"Huang2022_{apertures[x]}.hdf5"
+    output_directory = "../"
+
+    if not os.path.exists(output_directory):
+        os.mkdir(output_directory)
```
```suggestion
output_directory = "../"
if not os.path.exists(output_directory):
    os.mkdir(output_directory)

apertures = [10, 30, 50, 100]  # kpc

# Write a separate file for each aperture
for x, aperture in enumerate(apertures):
    output_filename = f"Huang2022_{aperture}kpc.hdf5"
```

EvgeniiChaikin commented on data/GalaxyStellarMassHaloMassBCG/conversion/convertKravstov2018.py:
```diff
@@ -0,0 +1,68 @@
+redshift = 0.15
+h = h_sim
+
+# Write a separate file for each aperture
+apertures = [30, 50, 100]
+for x in range(3):
+    output_filename = f"Kravstov2018_{apertures[x]}.hdf5"
+    output_directory = "../"
+
+    if not os.path.exists(output_directory):
+        os.mkdir(output_directory)
```
```suggestion
output_directory = "../"
if not os.path.exists(output_directory):
    os.mkdir(output_directory)

apertures = [30, 50, 100]  # kpc

# Write a separate file for each aperture
for x, aperture in enumerate(apertures):
    output_filename = f"Kravstov2018_{aperture}kpc.hdf5"
```

EvgeniiChaikin commented on data/GalaxyStellarMassHaloMassBCG/conversion/convertLin2017.py:
```diff
@@ -0,0 +1,81 @@
+plot_as = "points"
+redshift = 0.7
+redshift_lower = 0.0
+redshift_upper = 1
```
```suggestion
redshift_upper = 1.0
```

EvgeniiChaikin commented on data/GalaxyStellarMassHaloMassBCG/conversion/convertKravstov2018.py:
```diff
@@ -0,0 +1,68 @@
+bibcode = "2018AstL...44....8K"
+name = "BCG stellar mass-halo mass relation at z=0.15"
+plot_as = "points"
+redshift = 0.15
```
Do you also want to provide `redshift_lower` and `redshift_upper`, similar to how you do it in the other scripts?

EvgeniiChaikin reviewed (commented):
Hi @FHusko,

I added a couple of small comments in the code. 

A few more comments, 

- Do all these papers define M200 in the same way? And how is M200 exactly defined? Is it M200crit, mean, according to Bryan Norman (1998), or something else?
- It is not clear whether stellar masses are 2D or deprojected. E.g. Gonzalez et al (2013) provide both 2D and 3D masses.

EvgeniiChaikin commented:
Lastly, I'd also move the `comment` variable inside the loop and make it an f-string such that if, say, a file contains the data measured in 50 kpc apertures, then the comment would say so (instead of current line `"Stellar masses were measured in 10, 50 and 100 kpc apertures`, which makes we think that the file contains data measured in multiple apertures.).

FHusko commented:
Thanks Evgenii, I implemented your latest comments. As for halo masses: all that use 200 or 500 define them using critical densities. I now specify that. One paper uses the overdensity definition from Bryan & Norman (1998), I also specify that. 

We agreed with Joop to convert these all to M200_crit for reasons of having them all on the same plot, rather than looking at too many plots.

FHusko commented:
I added some small comments for the Golden-Marx et al. observations. They use a mean overdensity rather than a critical overdensity definition. I do convert to account for this in the data already, but forgot that I do. I have added details about that in the descriptions.

EvgeniiChaikin reviewed (approved):
Thanks, @FHusko! The scripts look ready to me; I am going to merge them.

EvgeniiChaikin commented:
Just for the purpose of keeping full records in this PR, this is how these data look


![stellar_mass_halo_mass_M200_all_100](https://user-images.githubusercontent.com/20153933/236076040-ebf38083-8095-404b-9044-b98ae002138d.png)
